dsl.md

```diff
@@ -0,0 +1,96 @@
+
+The DSL also embraces the principle of implicit default behaviors, sparing authors from unnecessary repetition and enhancing the conciseness of workflow definitions. For instance, default settings alleviate the burden of explicitly defining every detail, streamlining the workflow design process. Furthermore, the DSL allows both inline declaration of components or the creation of reusable elements, granting flexibility in workflow composition. This flexibility allows developers to seamlessly integrate inline task definitions without imposing rigid structural requirements.
+
+Moreover, the DSL eschews strong-typed enumerations wherever feasible, fostering extensibility and adaptability across different runtime environments. While maintaining portability is crucial, the DSL prioritizes customization options for extensions and runtimes, enabling tailored implementations to suit diverse use cases. Additionally, the DSL favors universally understood terms over technical jargon, enhancing accessibility and comprehension for a broader audience.
+
+- Embrace linguistic fluency for enhanced readability and understanding.
+- Utilize imperative verbs to convey actions directly and clearly.
+- Employ implicit default behaviors to reduce redundancy and streamline workflow definitions.
+- Enable the declaration and effortless import of shared components by supporting external references
+- Encourage the declaration of components inline for situations where reusability is unnecessary, prioritizing ease of use in such cases.
+- Prioritize flexibility over strong-typed enumerations for enhanced extensibility.
+- Opt for universally understood terms to improve accessibility and comprehension.
+
+## Concepts
+
+### Workflow
+
+A Serverless Workflow is a sequence of specific [tasks](#tasks) that are executed in a predetermined order. By default, this order follows the declaration sequence within the workflow definition. Workflows are designed to automate processes and orchestrate various serverless functions and services. 
+
+Workflows can be triggered in different ways: upon request, scheduled using CRON expressions, or initiated upon correlation with specific events. 
+
+Additionally, workflows may optionally accept inputs and produce outputs, allowing for data processing and transformation within the workflow execution.
+
+Workflows in the Serverless Workflow DSL can exist in several phases, each indicating the current state of the workflow execution. These phases include:
+
+| Phase |	Description |
+| --- | --- |
+| `Pending` |	The workflow has been initiated and is pending execution. |
+| `Running` |	The workflow is currently in progress. |
+| `Suspended` |	The workflow execution has been paused or halted temporarily. |
```

Comment:
I actually prefer the name `waiting` because the workflow is waiting for something to happen (tipically the reception of the event).

Reply:
yes, thats it, I think waiting reflect better the real state of the workflow

Reply:
In this case, the name is required, as it identifies a task amongst others. For/Do does not require name cause it defines a single task.